packages/react-app-revamp/hooks/useProposalVotes/index.ts

```diff
@@ -84,30 +101,15 @@ export function useProposalVotes(id: number | string) {
         args: id,
       });
 
-      await Promise.all(
-        list.map(async (userAddress: string) => {
-          const data = await readContract({
-            //@ts-ignore
-            ...contractConfig,
-            functionName: "proposalAddressVotes",
-            args: [id, userAddress],
-            chainId,
-          });
-
-          const author = await fetchEnsName({
-            address: userAddress,
-            chainId: wagmiChain.mainnet.id,
-          });
-          setVotesPerAddress({
-            address: userAddress,
-            value: {
-              displayAddress: author ?? shortenEthereumAddress(userAddress),
-              //@ts-ignore
-              votes: data?.forVotes ? data?.forVotes / 1e18 - data?.againstVotes / 1e18 : data / 1e18,
-            },
-          });
-        }),
-      );
+      // Pagination
+      const totalPagesPaginationVotes = Math.ceil(list?.length / VOTES_PER_PAGE);
+      setTotalPagesPaginationVotes(totalPagesPaginationVotes);
+      setCurrentPagePaginationVotes(0);
+      //@ts-ignore
+      const paginationChunks = arrayToChunks(list, VOTES_PER_PAGE);
+      setTotalPagesPaginationVotes(paginationChunks.length);
+      setIndexPaginationVotesPerId(paginationChunks);
+      if (list.length > 0) await fetchVotesPage(0, paginationChunks[0], paginationChunks.length);
```

Comment:
Would it be possible to check if the user's wallet is connected and if they have voted, and if so always make sure to have them be in the first chunk? I think this would be really helpful for people to be able to easily see what their current vote on a proposal is.